src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -1227,4 +1262,239 @@ const char* netName(FrNet* net)
+      painter.setPen(layer);
+      painter.setBrush(layer);
+    }
+    else {
+      painter.setPen(painter.cyan);
+      painter.setBrush(painter.cyan);
+    }
+    painter.setPenWidth(700);
+    painter.drawLine(x1, y1, x2, y2);
+  }
+
+}
+void FastRouteRenderer::drawTreeEdges(gui::Painter& painter){
+
+  int lastL;
+  for (int edgeID = 0; edgeID < 2 * deg_ - 3; edgeID++) {
+
+    TreeEdge* treeEdge = &(treeEdges_[edgeID]);
+
+    if (treeEdge->len > 0) {
+      int routeLen = treeEdge->route.routelen;
+      const std::vector<short>& gridsX = treeEdge->route.gridsX;
+      const std::vector<short>& gridsY = treeEdge->route.gridsY;
+      const std::vector<short>& gridsL = treeEdge->route.gridsL;
+      int lastX = w_tile_ * (gridsX[0] + 0.5) + x_corner_;
+      int lastY = h_tile_ * (gridsY[0] + 0.5) + y_corner_;
+
+      if (is3DVisualization_) 
+        lastL = gridsL[0];
+      
+      for (int i = 1; i <= routeLen; i++) {
+        const int xreal = w_tile_ * (gridsX[i] + 0.5) + x_corner_;
+        const int yreal = h_tile_ * (gridsY[i] + 0.5) + y_corner_;
+
+        if (is3DVisualization_) {
+          drawLineObject(lastX, lastY, lastL + 1, xreal, yreal, gridsL[i] + 1, painter);
+          lastL = gridsL[i];
+        }
+        else {
+          drawLineObject(lastX, lastY, -1, xreal, yreal, -1, painter);  // -1 to 2D Trees
+        }
+        lastX = xreal;
+        lastY = yreal;
+      }
+    }
+  }
+}
+void FastRouteRenderer::drawCircleObjects(gui::Painter& painter){
+
+  for (int i = 0; i < num_pins_; i++) {
+    const int xreal = w_tile_ * (pinX_[i] + 0.5) + x_corner_;
+    const int yreal = h_tile_ * (pinY_[i] + 0.5) + y_corner_;
+    
+    odb::dbTechLayer* layer = tech_->findRoutingLayer(pinL_[i] + 1);
+    painter.setPen(layer);
+    painter.setBrush(layer);
+    painter.setPenWidth(700);
+    painter.drawCircle(xreal, yreal,1500);
+  }
+}
+
+void FastRouteRenderer::drawObjects(gui::Painter& painter)
+{
+  if (treeStructure_ == 1) {
+  
+    const int deg = stree_.deg;
+    for (int i = 0; i < 2 * deg - 2; i++) {
+      const int x1 = w_tile_ * (stree_.branch[i].x + 0.5) + x_corner_;
+      const int y1 = h_tile_ * (stree_.branch[i].y + 0.5) + y_corner_;
+      const int n = stree_.branch[i].n;
+      const int x2 = w_tile_ * (stree_.branch[n].x + 0.5) + x_corner_;
+      const int y2 = h_tile_ * (stree_.branch[n].y + 0.5) + y_corner_;
+      const int len = abs(x1-x2)+abs(y1-y2);
+      if (len > 0) {
+          painter.setPen(painter.white);
+          painter.setBrush(painter.white);
+          painter.setPenWidth(700);
+          painter.drawLine(x1,y1,x2,y2);
+      }
+    }
+    
+    drawCircleObjects(painter);
+  }
+  else if (treeStructure_ == 2) {
+    
+    drawTreeEdges(painter);
+    
+    drawCircleObjects(painter);
+  }
+}
+
+////////////////////////////////////////////////////////////////
+void FastRouteCore::createDebug()
+{
+  debug_ = new DebugSetting();
+}
```

Comment:
I only see a caller in the ctor so you might as well just move this one line there.